external/fv3fit/fv3fit/_shared/packer.py

```diff
@@ -234,28 +350,68 @@ def count_features(
 
     Args:
         quantity_names: names of variables to include in the count
-        dataset: a dataset containing the indicated variables
+        dataset: a dataset containing the indicated variables,
+            dimensionality must match value of is_3d
         sample_dim_name: dimension to treat as the "sample" dimension, any other
             dimensions are treated as a "feature" dimension.
+        is_3d: pass as True if the dataset contains a time dimension
```

Comment:
Yes, the for loop below is vestigial from when I was trying to detect this automatically, but it can't work for all cases (particularly, when all values are scalars and the data is 3D it's indistinguishable from a 3D dataset). I should delete the for loop.

Reply:
Once you refactor to `ArrayPacker3d` it should be possible to delete this dispatching function and call the `_count_features_2d/3d` helpers directly.